plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -0,0 +1,328 @@
+                columnStatistics.putIfAbsent(id, new ColumnStatistics(type, typeManager, lowerBound, upperBound));
+                columnStatistics.get(id).updateMinMax(lowerBound, upperBound);
+            }
+        }
+
+        private static Map<Integer, Object> convertBounds(Map<Integer, Type.PrimitiveType> idToTypeMapping, Map<Integer, ByteBuffer> idToMetricMap)
+        {
+            if (idToMetricMap == null) {
+                return null;
+            }
+            ImmutableMap.Builder<Integer, Object> map = ImmutableMap.builder();
+            idToMetricMap.forEach((id, value) -> {
+                Type.PrimitiveType type = idToTypeMapping.get(id);
+                verify(type != null, "No type for column id %s, known types: %s", id, idToTypeMapping);
+                map.put(id, Conversions.fromByteBuffer(type, value));
+            });
+            return map.build();
+        }
+    }
+
+    private static class ColumnStatistics
+    {
+        private final io.trino.spi.type.Type type;
+        private final TypeManager typeManager;
+        private Optional<Object> min;
+        private Optional<Object> max;
+
+        public ColumnStatistics(io.trino.spi.type.Type type, TypeManager typeManager, Object initialMin, Object initialMax)
+        {
+            this.type = requireNonNull(type, "type is null");
+            this.typeManager = requireNonNull(typeManager, "typeManager is null");
+            this.min = Optional.ofNullable(initialMin);
+            this.max = Optional.ofNullable(initialMax);
+        }
+
+        public Optional<Object> getMin()
+        {
+            return min;
+        }
+
+        public Optional<Object> getMax()
+        {
+            return max;
+        }
+
+        /**
+         * @param lowerBound Trino encoded lower bound value from a file
+         * @param upperBound Trino encoded upper bound value from a file
+         */
+        public void updateMinMax(Object lowerBound, Object upperBound)
+        {
+            // Update the stats, as long as they haven't already been invalidated
+            if (min.isPresent()) {
+                if (lowerBound == null) {
+                    min = Optional.empty();
+                }
+                else if (compareTrinoValue(lowerBound, min.get()) < 0) {
+                    min = Optional.of(lowerBound);
+                }
+            }
+
+            if (max.isPresent()) {
+                if (upperBound == null) {
+                    max = Optional.empty();
+                }
+                else if (compareTrinoValue(upperBound, max.get()) > 0) {
+                    max = Optional.of(upperBound);
+                }
+            }
+        }
+
+        private long compareTrinoValue(Object value, Object otherValue)
+        {
+            MethodHandle comparisonMethod = typeManager.getTypeOperators().getComparisonUnorderedLastOperator(type, simpleConvention(FAIL_ON_NULL, NEVER_NULL, NEVER_NULL));
```

Comment:
Evaluate in the constructor.

(and make sure the `ColumnStatistics::ColumnStatistics` is invoked only when necessary)